report/5.Implementation/MovementModule.tex

```diff
@@ -0,0 +1,117 @@
+
+\begin{lstlisting}[language=CSharp,label={lst:InitMotor},caption={init\_motor method from movement.c}]
+bool init_motor(uint8_t motor_id, char orientation, uint16_t speed){
+  if(orientation == 'x'){
+    x_motor = motor_id;
+    x_motor_speed = speed;
+    ecrobot_set_motor_speed(x_motor, 0);
+    nxt_motor_set_count(x_motor, 0);
+    current_location.x = ecrobot_get_motor_rev(motor_id);
+    return true;
+  }
+  else if(orientation == 'y'){
+    y_motor = motor_id;
+    y_motor_speed = speed;
+    ecrobot_set_motor_speed(y_motor, 0);
+    nxt_motor_set_count(y_motor, 0);
+    current_location.y = ecrobot_get_motor_rev(motor_id);
+    return true;
+  }
+  return false;
+}
+\end{lstlisting}
+As it can be seen on Snippet~\ref{lst:DeviceInit} line 2 and 3 the parameter for the motor\_id is the port id on the NXT device. 
+The character for orientation of the motor is used to distinguish between the two motors as they are connected to the NXT.
+The speed is set to an initialization value of $20$ which can be any given value between $0$ and $100$.
+
+\subsection{The motion}
+Before any meaningful motion can be made it is essential to know where the turret is currently aiming and where the target is located. 
+Without this information any motion will result in a meaningless motion and should therefore be avoided. 
+
+When the machine intelligence system has found and located the target it will send the information to the NXT and the NXT will then start to aim at the target. 
+The target location is stored in an struct containing all relevant information about the target. 
+The move\_motors method is responsible for the actual activation of the individual motors.
+The method can be seen on Snippet~\ref{lst:MoveMotors} and it will activate the motors according to the targets location and where it is currently aiming. 
+\begin{lstlisting}[language=CSharp,caption={move\_motors method from movement.c},label={lst:MoveMotors}]
+void move_motors(){
+  current_location.x = ecrobot_get_motor_rev(x_motor);
+  current_location.y = ecrobot_get_motor_rev(y_motor);
+
+  if(current_location.x > target_location.x){
+    ecrobot_set_motor_speed(x_motor, -x_motor_speed);
+  }
+  else if(current_location.x < target_location.x){
+    ecrobot_set_motor_speed(x_motor, x_motor_speed);
+  }
+  if(current_location.y > target_location.y){
+    ecrobot_set_motor_speed(y_motor, -y_motor_speed);
+  }
+  else if(current_location.y < target_location.y){
+    ecrobot_set_motor_speed(y_motor, y_motor_speed);
+  }
+}
+\end{lstlisting}
+Once the motors has engaged and the turret is aiming at the target, the motion should be stopped. 
+If power is simply removed from the motors the motor will loos their stiffness and the entire turret will power down. 
+To avoid this the motors has a function to simply set the speed of the motors to $0$ resulting in the motors locking up since they will actively try to counter act any motion. 
```

Comment:
```suggestion
To avoid this, the motors have a function to simply set the speed of the motors to $0$, resulting in the motors locking up since they will actively try to counteract any motion. 
```